python_bindings/src/PyDerivative.cpp

```diff
@@ -0,0 +1,20 @@
+#include "PyDerivative.h"
+
+namespace Halide {
+namespace PythonBindings {
+
+void define_derivative(py::module &m) {
+    auto derivative_class = py::class_<Derivative>(m, "Derivative")
+        .def("__getitem__", [](const Derivative &d, const Func &func) {return d(func);})
```

Comment:
The equivalent method in C++ has two optional arguments. Do we want to support those here?

Reply:
Addressed. The second optional argument in the C++ function is actually unnecessary and should be removed.

Reply:
Actually I am wrong. The bounded argument was there for easier scheduling. I'll add it.